### DISCUSSION TITLE
Directory suggestion and CTRL+R not working as expected

### DISCUSSION THREAD

**Discussion by nstoik** [marlonrichert/zsh-autocomplete#481] [Q&A] [answered]
Thanks for making this plugin, I like the way it looks but I am having a few problems trying to use it. I am not sure exactly how to phrase this question or if I am using the correct terminology, but I am unable to get zsh-autocomplete to function as expected or how the README shows. I wasn't able to find any solutions in other discussions or issues.

I use oh-my-zsh and created a minimal configuration on my computer with only the zsh-autocomplete plugin enabled and the behavior is the same. The .zshrc used is [here](https://github.com/nstoik/dotfiles/blob/zsh-autocomplete-issue/shells/zsh/zshrc).

1. First off, `CTRL+R` does not open Live History mode. When pressing `CTRL+R` nothing happens. Pressing `UP` does open up the Live History mode. When disabling the zsh-autocomplete plugin, `CTRL+R` does work as expected.
2. While the directory suggestion shows when at my home directory, it doesn't show when changing to another directory.

![image](https://user-images.githubusercontent.com/11414257/188786704-9c2a9d01-8dc0-433e-9ab7-c4908259b326.png)

After changing into a different directory, no directory suggestions or command autocomplete is provided. Pressing `TAB` causes the suggestions to briefly flash across the screen, but then they immediately disappear.

![image](https://user-images.githubusercontent.com/11414257/188787162-16edaccd-99ab-4e66-a6ad-fe001b694a17.png)

Let me know if there are any suggestions to try or if you need any further information.

**Comment by marlonrichert** [comment, marked as answer]
> First off, `CTRL+R` does not open Live History mode. When pressing `CTRL+R` nothing happens.

With your minimal config, what do you get from `bindkey '^R'` and `zle -lL history-incremental-search-backward`?

> After changing into a different directory, no directory suggestions or command autocomplete is provided. 

What do you get from `ls` in that dir?

> Pressing TAB causes the suggestions to briefly flash across the screen, but then they immediately disappear.

When this happens, what output do you get in the files `$_autocomplete__log`, `$_autocomplete__log_async` and `$_autocomplete__log_pty`?

**Reply by nstoik** [reply]
> With your minimal config, what do you get from bindkey '^R' and zle -lL history-incremental-search-backward?

Those commands show  `"^R" history-incremental-search-backward` and `zle -N history-incremental-search-backward .autocomplete.async.history-incremental-search` as output

> What do you get from ls in that dir?

`ls` shows the subfolders present. You can see in the recording below the suggestions flash on the screen when I press `TAB`

> When this happens, what output do you get in the files $_autocomplete__log, $_autocomplete__log_async and $_autocomplete__log_pty?

For the log files, the first file has `.autocomplete.async.complete.fd-widget.inner:25: completion cannot be used recursively (yet)` repeated multiple times, and the other two files are empty.

Here is a recording of the steps to hopefully visualize it better. [![asciicast](https://asciinema.org/a/519861.svg)](https://asciinema.org/a/519861)

**Reply by marlonrichert** [reply]
Can you open a bug report and carefully follow the instructions in the template? Then I can have closer look.

**Reply by nstoik** [reply]
As I was going through the steps to create the bug report the minimal steps in the template worked!!

```zsh
cd $(mktemp -d)
git clone --depth 1 -- https://github.com/marlonrichert/zsh-autocomplete.git
> .zshrc <<EOF
PS1='%# ' PS2= RPS2='%^'; setopt transientrprompt interactivecomments
source ~/zsh-autocomplete/zsh-autocomplete.plugin.zsh
EOF
env -i HOME=$PWD PATH=$PATH TERM=$TERM ${TERMINFO:+TERMINFO=$TERMINFO} zsh -d
```

After running those steps, `CTRL+R` and `TAB` work in the temp terminal. 

So I slimmed down my `.zshrc` file even further but the normal terminal is still behaving the same. Do you want me to submit a bug report still? Any suggestion what to try next? I am wondering if it could be something with oh-my-zsh, but I am unsure how to debug further.

**Reply by nstoik** [reply]
Some success with this! To try figure out what was causing the problem, I played around with building the configuration up step by step until the problem started happening.

It ended up being something with `oh-my-zsh`. I don't know exactly what but whenever that was added to the `.zshrc` file, the problems started happening.

I ditched `oh-my-zsh` and switched to `zsh-snap` and everything seems to be working now.

**Reply by marlonrichert** [reply]
Thanks, that’s good to know. I’ll test with OMZ and see what’s the problem.
